hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterWrapper.java

```diff
@@ -200,19 +202,34 @@ public boolean isPolling() {
         return adapter instanceof PollingProtocolAdapter;
     }
 
+    public boolean isBatchPolling() {
+        return adapter instanceof BatchPollingProtocolAdapter;
+    }
+
     private void startPolling(
             final @NotNull ProtocolAdapterPollingService protocolAdapterPollingService,
             final @NotNull EventService eventService) {
-        if (isPolling()) {
-            log.debug("Schedule polling for protocol adapter with id '{}'", getId());
-            final PerAdapterSampler sampler = new PerAdapterSampler(this, getNorthboundMappings(), eventService, tagManager);
+
+        if (isBatchPolling()) {
+            log.debug("Schedule batch polling for protocol adapter with id '{}'", getId());
+            final PerAdapterSampler sampler =
+                    new PerAdapterSampler(this, getNorthboundMappings(), eventService, tagManager);
             protocolAdapterPollingService.schedulePolling(sampler);
         }
+
+        if (isPolling()) {
+            getNorthboundMappings().forEach(adapterSubscription -> {
+                log.debug("Schedule polling for protocol adapter with id '{}'", getId());
+                final PerContextSampler sampler =
+                        new PerContextSampler(this, adapterSubscription, eventService, tagManager);
+                protocolAdapterPollingService.schedulePolling(sampler);
+            });
+        }
```

Comment:
What if it is neither? Unlikely but their might be an implementation bug hiding when non of the both is true.

Reply:
neither is a valid case f.e. opcua. If there is nothing to poll from, we dont start polling. but yeah we can miss implementations here if we add a new one.